clients/telegrafs/telegrafs.go

```diff
@@ -0,0 +1,187 @@
+package telegrafs
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.TelegrafsApi
+}
+
+type telegrafPrintOpts struct {
+	graf  *api.Telegraf
+	grafs *api.Telegrafs
+}
+
+type ListParams struct {
+	clients.OrgParams
+	Id string
+}
+
+func (c Client) List(ctx context.Context, params *ListParams) error {
+	if !params.OrgID.Valid() || params.Id == "" {
+		return errors.New("at least one of org, org-id, or id must be provided")
+	}
+
+	if params.Id != "" {
+		id, err := influxid.IDFromString(params.Id)
+		if err != nil {
+			return err
+		}
+
+		telegraf, err := c.GetTelegrafsID(ctx, id.String()).Execute()
+		if err != nil {
+			return err
+		}
+
+		return c.printTelegrafs(telegrafPrintOpts{graf: &telegraf})
+	}
+
+	telegrafs, err := c.GetTelegrafs(ctx).OrgID(params.OrgID.String()).Execute()
+	if err != nil {
+		return err
+	}
+
+	return c.printTelegrafs(telegrafPrintOpts{grafs: &telegrafs})
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc string
+	File string
+	Name string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	cfg, err := c.readConfig(params.File)
+	if err != nil {
+		return err
+	}
+
+	orgID := params.OrgID.String()
+
+	newTelegraf := api.TelegrafRequest{
+		Name:        &params.Name,
+		Description: &params.Desc,
+		Config:      &cfg,
+		OrgID:       &orgID,
+	}
+
+	graf, err := c.PostTelegrafs(ctx).TelegrafRequest(newTelegraf).Execute()
+	if err != nil {
+		return err
+	}
+
+	return c.printTelegrafs(telegrafPrintOpts{graf: &graf})
+}
+
+type RemoveParams struct {
+	clients.OrgParams
+	Ids []string
+}
+
+func (c Client) Remove(ctx context.Context, params *RemoveParams) error {
+	for _, rawID := range params.Ids {
+		id, err := influxid.IDFromString(rawID)
```

Comment:
I don't think there's much point in these conversions since we're just calling `String()` on them below. The server will reject any malformed IDs when the request is sent.